servant/src/Servant/API/ContentTypes.hs

```diff
@@ -219,9 +226,10 @@ instance AllMime '[] where
     allMime _ = []
 
 instance (Accept ctyp, AllMime ctyps) => AllMime (ctyp ': ctyps) where
-    allMime _ = (contentType pctyp):allMime pctyps
-      where pctyp  = Proxy :: Proxy ctyp
-            pctyps = Proxy :: Proxy ctyps
+    allMime _ = NE.toList (contentTypes pctyp) ++ allMime pctyps
+      where
+        pctyp  = Proxy :: Proxy ctyp
+        pctyps = Proxy :: Proxy ctyps
```

Comment:
I guess there the name is choose to resemble type level variable:
```hs
pctyp ~ ctyp
pctyps ~ ctyps
```
would
```hs
ctypProxy = Proxy :: Proxy ctyp
```
be good middleground?